cdap-app-fabric/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -81,4 +127,205 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+   * @throws IOException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundle/{uuid}")
+  public void getSupportBundleByUUID(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.GET);
+    File baseDirectory = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR));
+    File uuidFile = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR), uuid);
+    if (!baseDirectory.exists()) {
+      responder.sendString(HttpResponseStatus.OK, "No content in Support Bundle.");
+      return;
+    }
+    if (!uuidFile.exists()) {
+      responder.sendString(HttpResponseStatus.OK, String.format("No such uuid %s in Support Bundle.", uuid));
+      return;
+    }
+    JsonArray statusJsonArray = new JsonArray();
+    JsonObject singleBundleJson = bundleService.getSingleBundleJson(uuidFile);
+    statusJsonArray.add(singleBundleJson);
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(statusJsonArray));
+  }
+
+  /**
+   * Delete the specific bundle by using the uuid
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws IOException is thrown if status.json is not found
+   */
+  @DELETE
+  @Path("/support/bundle/{uuid}")
+  public void deleteSupportBundle(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.DELETE);
+    File baseDirectory = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR));
+    File uuidFile = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR), uuid);
+    if (!baseDirectory.exists()) {
+      responder.sendString(HttpResponseStatus.OK, "No content in Support Bundle.");
+      return;
+    }
+    if (!uuidFile.exists()) {
+      responder.sendString(HttpResponseStatus.OK, String.format("No such uuid %s in Support Bundle.", uuid));
+      return;
+    }
+    bundleService.deleteOldFolders(uuidFile);
+    responder.sendString(HttpResponseStatus.OK, String.format("Successfully deleted bundle %s", uuid));
+  }
+
+  /**
+   * Get the specific bundle file list or certain file content
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws IOException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundle/{uuid}/files")
+  public void getSupportBundleFile(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid,
+                                   @DefaultValue("") @QueryParam("folder-name") String folderName,
+                                   @DefaultValue("") @QueryParam("data-file-name") String dataFileName)
+    throws IOException {
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.GET);
```

Comment:
This should probably use enforce instead of enforceOnParent and enforce upon the specific SupportBundleId instead of on the parent.